src/components/screens/addedappid.tsx

```diff
@@ -1,8 +1,98 @@
-import React from "react";
-import { Text, View, Image, TouchableOpacity } from "react-native";
+import React, { useState, useEffect } from "react";
+import { Text, View, Image, TouchableOpacity, ScrollView, Alert } from "react-native";
+import db from "../../app/db";
+import * as Babel from "@babel/standalone";
 
-function AppDetail({ appData, setPage, panther, isActive, setIsActive }) {
-  const { appId, appName, appDesc } = appData;
+function AppDetail({ appData, setPage, panther, isActive }) {
+  const { appId } = appData;
+  const [codeResult, setCodeResult] = useState(null);
+  const [codeError, setCodeError] = useState(null);
+
+  const { isLoading, error, data } = db.useQuery({
+    appslist: {
+      $: {
+        where: {
+          id: appId,
+        },
+      },
+    },
+  });
+
+  const app = data?.appslist?.[0];
+
+  const displayAppName = app?.appname;
+  const displayAppDesc = app?.appdesc;
+  const displayCode = app?.code;
+
+  useEffect(() => {
+    if (displayCode) {
+      try {
+        const cleanCode = displayCode.trim();
+        
+        const transformed = Babel.transform(cleanCode, {
```

Comment:
I would move all of this code into a separate function, to keep things a bit cleaner. I would also ask AI to help you understand what is happening here